intellij-toml/src/main/kotlin/org/toml/ide/folding/TomlFoldingBuilder.kt

```diff
@@ -0,0 +1,57 @@
+package org.toml.ide.folding;
+
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingBuilderEx
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.toml.lang.psi.*
+
+class TomlFoldingBuilder : FoldingBuilderEx(), DumbAware {
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<FoldingDescriptor> {
+        if (root !is TomlFile) return emptyArray()
+
+        val visitor = TomlFoldingVisitor()
+        PsiTreeUtil.processElements(root) {
+            visitor.visit(it)
+            true
+        }
+        return visitor.folds.toTypedArray()
+    }
+
+    override fun getPlaceholderText(node: ASTNode): String? = when (node) {
+        is TomlInlineTable -> "{...}"
```

Comment:
Hm, it looks suspocious because `PsiElement` may not implement `ASTNode` - it depends on concrete implementation.
So if we change psi implementation for TOML, it may break folding.
I see two ways to fix it:
- use `node.psi` to get proper psi element for node
- use `node.elementType` instead of PSI classes

I prefer the second way